packages/smartcontracts/src/tests/BridgeProxy.proxyBehaviour.spec.ts

```diff
@@ -0,0 +1,151 @@
+
+describe('Test Behaviour related to proxy', () => {
+  describe('Test Proxy Upgradeability', () => {
+    it('The smart contract retains the values of state variables after upgrading', async () => {
+      const { testToken, proxyBridge, defaultAdminSigner } = await loadFixture(deployContracts);
+      const accounts = await ethers.provider.listAccounts();
+
+      const NewImplementation = await ethers.getContractFactory('NewImplementation');
+      const newImplementation = await NewImplementation.deploy();
+      await newImplementation.deployed();
+
+      await testToken.mint(proxyBridge.address, toWei('100'));
+      await proxyBridge.addSupportedTokens(testToken.address, toWei('15'));
+
+      const eip712Types = {
+        CLAIM: [
+          { name: 'to', type: 'address' },
+          { name: 'amount', type: 'uint256' },
+          { name: 'nonce', type: 'uint256' },
+          { name: 'deadline', type: 'uint256' },
+          { name: 'tokenAddress', type: 'address' },
+        ],
+      };
+      const domainData = {
+        name: 'CAKE_BRIDGE',
+        version: '0.1',
+        chainId: 1337,
+        verifyingContract: proxyBridge.address,
+      };
+      const eip712Data = {
+        to: accounts[2],
+        amount: toWei('10'),
+        nonce: 0,
+        deadline: ethers.constants.MaxUint256,
+        tokenAddress: testToken.address,
+      };
+
+      const signature = await defaultAdminSigner._signTypedData(domainData, eip712Types, eip712Data);
+      proxyBridge.claimFund(accounts[2], toWei('10'), 0, ethers.constants.MaxUint256, testToken.address, signature);
+
+      // upgrade to the new contract
+      await proxyBridge.upgradeTo(newImplementation.address);
+      // check the implementation slot
+      expect(
+        `0x${(
+          await ethers.provider.getStorageAt(
+            proxyBridge.address,
+            '0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc',
+          )
+        ).substring(26)}`,
+      ).to.equal(newImplementation.address.toLowerCase());
+      // check admin role and other storage slots
+      const upgradedProxyBridge = NewImplementation.attach(proxyBridge.address);
+      expect(
+        await upgradedProxyBridge.hasRole(
+          '0x0000000000000000000000000000000000000000000000000000000000000000',
+          defaultAdminSigner.address,
+        ),
+      ).to.equal(true);
+      expect(await upgradedProxyBridge.eoaAddressToNonce(accounts[2])).to.equal(1);
+      expect(await upgradedProxyBridge.supportedTokens(testToken.address)).to.equal(true);
+      expect(await upgradedProxyBridge.relayerAddress()).to.equal(defaultAdminSigner.address);
+      expect((await upgradedProxyBridge.tokenAllowances(testToken.address)).dailyAllowance).to.equal(toWei('15'));
+      expect(await upgradedProxyBridge.transactionFee()).to.equal(30);
+    });
+  });
+
+  describe('Test different situations of calling the proxy contract', () => {
+    let abiCoder: any;
```

Comment:
```suggestion
    let abiCoder: AbiCoder;
```

You'll need to make the appropriate import as well.